server/pulp/server/async/app.py

```diff
@@ -72,7 +166,9 @@ def initialize_worker(sender, instance, **kwargs):
     initialization.initialize()
 
     # Delete any potential old state
-    tasks._delete_worker(sender, normal_shutdown=True)
+    existing_worker = Worker.objects(name=sender).first()
+    if existing_worker:
```

Comment:
Responding to 3 comments ago. Yes we can't call delete here for those reasons. I'm using deductive reasons to point out that reading in database state and having conditional logic on it is a pattern for a race condition.